aibtc-v1/chat.py

```diff
@@ -0,0 +1,171 @@
+import streamlit as st
+import logging
+from typing import Dict, Any
+from utils.session import init_session_state
+
+# Initialize session state
+init_session_state()
+
+# Set up Streamlit page
+st.set_page_config(page_title="CrewAI Chatbot", layout="centered")
+
+# Constants
+AVAILABLE_CREWS: Dict[str, Dict[str, Any]] = st.session_state.crew_mapping
+
+
+def load_custom_styles():
+    st.markdown(
+        """
+    <style>
+    /* Custom styles here */
+    </style>
+    """,
+        unsafe_allow_html=True,
+    )
+
+
+def generate_initial_welcome_message() -> str:
+    message = "Welcome to CrewAI Chatbot! I can help you run different CrewAIs. Here are the available options:\n\n"
+    message += "\n".join(
+        f"- **{name}**: {info['description']}" for name, info in AVAILABLE_CREWS.items()
+    )
+    message += "\n\nYou can ask me more about any of these or tell me what you want to do, and I'll recommend a CrewAI to run."
+    return message
+
+
+INITIAL_WELCOME_MESSAGE = generate_initial_welcome_message()
+
+
+def initialize_session_state():
+    if st.session_state.messages == []:
+        st.session_state.messages = [
+            {"role": "Bot", "content": INITIAL_WELCOME_MESSAGE}
+        ]
+    if "crew_selected" not in st.session_state:
+        st.session_state.crew_selected = None
+    if "conversation_context" not in st.session_state:
+        st.session_state.conversation_context = "awaiting_selection"
+    if "crew_step_container" not in st.session_state:
+        st.session_state.crew_step_container = st.empty()
+    if "crew_task_container" not in st.session_state:
+        st.session_state.crew_task_container = st.empty()
+    if "crew_step_callback" not in st.session_state:
+        st.session_state.crew_step_callback = []
+    if "crew_task_callback" not in st.session_state:
+        st.session_state.crew_task_callback = []
+
+
+def add_to_chat(speaker: str, message: str):
+    st.chat_message(speaker).write(message)
+    st.session_state.messages.append({"role": speaker, "content": message})
+
+
+def handle_conversation(user_input: str) -> tuple:
```

Comment:
I wonder if we would benefit from using a `Crew` object here vs doing an LLM call - it changes what gets prompted and that could go either way.